examples/app/MonitorPanelPlot.ipynb

```diff
@@ -0,0 +1,398 @@
+    "\n",
+    "# Generate the cost schedule\n",
+    "def generate_cost_schedule(start_date, end_date, cost_change_date):\n",
+    "    cost_schedule = {}\n",
+    "    current_date = start_date\n",
+    "    cost_per_token = 0.01\n",
+    "    while current_date <= end_date:\n",
+    "        if current_date >= cost_change_date:\n",
+    "            cost_per_token = 0.005\n",
+    "        cost_schedule[current_date.date()] = cost_per_token\n",
+    "        current_date += timedelta(days=1)\n",
+    "    return cost_schedule\n",
+    "\n",
+    "def generate_user_usage_schedule(start_date, end_date, users):\n",
+    "    user_usage_schedule = []\n",
+    "    for user in users:\n",
+    "        current_date = start_date + timedelta(days=random.randrange(90))\n",
+    "        while current_date <= end_date:\n",
+    "            usage_periods = random.randint(1, 30)\n",
+    "            for _ in range(usage_periods):\n",
+    "                period_length_timedelta = timedelta(hours=random.randint(1, 24 * 7))\n",
+    "                rate = random.uniform(0.1, 10)\n",
+    "                user_usage_schedule.append((current_date, user, period_length_timedelta, rate))\n",
+    "                current_date += period_length_timedelta  # Increment current_date\n",
+    "                if current_date > end_date:\n",
+    "                    break\n",
+    "    return user_usage_schedule"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9d6a0318",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define our fake users\n",
+    "fake = Faker()\n",
+    "users = [fake.user_name() for _ in range(10)]\n",
+    "\n",
+    "# Read the file and generate prompts\n",
+    "with open('../../weave/testdata/t8.shakespeare.txt', 'r') as f:\n",
+    "    lines = f.read().split('\\n')\n",
+    "\n",
+    "# Define the time range\n",
+    "start_date = pd.to_datetime('2023-01-01', utc=True)\n",
+    "end_date = pd.to_datetime('2023-03-31', utc=True)\n",
+    "cost_change_date = pd.to_datetime('2023-02-15', utc=True)\n",
+    "\n",
+    "# Generate the schedules\n",
+    "version_schedule = generate_version_schedule(start_date, end_date)\n",
+    "latency_schedule = generate_latency_schedule(start_date, end_date)\n",
+    "cost_schedule = generate_cost_schedule(start_date, end_date, cost_change_date)\n",
+    "user_usage_schedule = generate_user_usage_schedule(start_date, end_date, users)\n",
+    "\n",
+    "# Helper function to generate a random completion\n",
+    "def generate_completion(prompt):\n",
+    "    words = prompt.split()\n",
+    "    completion = ' '.join(random.choices(words, k=int(len(words)* (random.random() + 0.1) * 10)))\n",
+    "    return completion\n",
+    "\n",
+    "#latency_schedule"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a672c782",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import tqdm\n",
```

Comment:
unenumerated dependency. maybe instead of adding to requirements we could do a `pip install` line at the beginning of the notebook to install the notebook's dependencies